### PR TITLE
fix apt::source to use global ppa

### PR DESCRIPTION
**Pull request by micah-uber**
Fix source to use ppa that is available for nginx which will also solve #929

### CODE REVIEW

File: manifests/package/debian.pp [hunk 1]
```diff
@@ -33,17 +33,13 @@
...
+        apt::ppa { 'ppa:nginx/stable':
+          ensure => present,
         }
       }
       'nginx-mainline': {
-        apt::source { 'nginx':
-          location => "http://nginx.org/packages/mainline/${distro}",
-          repos    => 'nginx',
-          key      => '573BFD6B3D8FBC641079A6ABABF5BD827BD9BF62',
+        apt::ppa { 'ppa:nginx/stable':
```
Comment (hunner) [inline manifests/package/debian.pp]: mainline is `ppa:nginx/development` according to https://launchpad.net/~nginx/+archive/ubuntu/development

**Comment by wyardley** [comment]
@micah-uber: thanks! I think it's still been working in the acceptance tests and in my quick functional tests, but spoke to folks in the community meeting, and it seems like moving to ppa is a fine way to go.
Would you mind resolving the test errors, resolving @hunner's feedback above, and switching to https (this will then preempt #939)

Generally, with voxpupuli modules, they want the commits commits to all be squashed to one, so if you can squash as well (interactive rebase / force-push), that would be great.

**Comment by wyardley** [comment]
A bunch of the failures have this error: `apt::ppa is not currently supported on Debian. at /home/travis/build/voxpupuli/puppet-nginx/spec/fixtures/modules/apt/manifests/ppa.pp:14 on node testing-worker-linux-docker-481429b0-3451-linux-10.prod.travis-ci.org`

**Comment by wyardley** [comment]
It sounds like ppa is Ubuntu specific.

**Comment by 3flex** [comment]
As @wyardley said PPAs are Ubuntu specific. I don't see a good reason to make this change, unless the PPA were only used on Ubuntu and the current method were still used for other Debian-derived distros, but even then I don't think it's a good idea to have the development or mainline versions come from any source other than nginx.org for consistency's sake.

**Comment by wyardley** [comment]
I'm going to close this one, though I agree we still do need to get a handle on things if there's a problem with signing key somewhere.

If you're seeing problems with the signing keys on new installs, can you open a new issue and provide `apt-cache policy nginx` output as well as the package related settings from your config, or else propose a new PR that reworks this in a way that preserves Debian support? I personally think it's probably not a good idea to use a different package install strategy for Ubuntu here.

**Pull request closed without merging**